packages/shared/src/components/modals/PostToSquadModal.tsx

```diff
@@ -57,9 +59,12 @@ function PostToSquadModal({
     buttonText: 'Done',
     preview,
   });
+  const { completeAction } = useActions();
 
   const shouldShowToggle = !isDismissed && !isSubscribed;
   const onPostSuccess = async (squadPost?: Post) => {
+    completeAction(ActionType.SquadFirstPost);
```

Comment:
@sshanzel @rebelchris what do you think of where should we actually finish this kind of action. 

Should we keep this kind of call from the frontend or we could potentially add logic to mutation on API side and then just call `invalidateQueries`( as we do for `sourceFeed` in this same callback) to refetch actions and show updated state in checklist?

For actions like extension download of course we have to do it from the frontend but for this one we could do it on the API since we already know action is completed there. I can see pros/cons for both ways, curious what you think.

Reply:
This is already handled in the BE (postAdded worker) though we might need to update/invalidate the cache from the frontend.

Reply:
I'll add cache invalidation then as mentioned above.